extensions/harvester/sakubeconfig.go

```diff
@@ -0,0 +1,98 @@
+package harvester
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	provv1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters"
+)
+
+const (
+	HarvesterProviderClusterLabel = "labelSelector=provider.cattle.io=harvester"
+)
+
+// KubeConfigOpts is configuration need to get a kubeconfig with SA
+type KubeConfigOpts struct {
+	CSIClusterRoleName string `yaml:"csiclusterRoleName" json:"csiclusterRoleName" default:"harvesterhci.io:csi-driver"`
+	ClusterRoleName    string `yaml:"clusterRoleName" json:"clusterRoleName" default:"harvesterhci.io:cloudprovider"`
+	Namespace          string `yaml:"namespace" json:"namespace" default:"default"`
+	SaName             string `yaml:"serviceAccountName" json:"serviceAccountName" default:"rancherCluster"`
+}
+
+// GetHarvesterSAKubeconfig generates a kubeconfig from the harvester cluster with a custom SA for the
+// cluster which will use the kubeconfig. This is typically used when creating a downstream cluster for
+// harvester that uses the harvester cloud provider.
+func GetHarvesterSAKubeconfig(client *rancher.Client, clusterName string) ([]byte, error) {
```

Comment:
do we need the client for any rbac related cases like a standard user for instance? if not we can just take in adminClient as the client is not used anywhere other than getting the adminClient in line 35.

Reply:
Good question, can we always assume it's an admin client?

Reply:
yeah sorry you guys are right, this can just be the client of the user who needs it.